### PR TITLE
Database replication strategies discussed at Percona Live 2019.

### PR DESCRIPTION
**Pull request by lieut-data**
This is a write-up of some of what I was exposed to at Percona Live 2019 concerning database replication strategies.

Amy, might I get your input on language used?

Since it's hard to review the images in context, here's a snapshot of what one looks like:

image

### CODE REVIEW

File: site/static/css/styles.css [hunk 1]
```diff
@@ -3410,7 +3410,7 @@ table>tbody>tr:nth-of-type(odd) {
 
 .doc-content img {
     margin: 1em 0;
-    max-width: 800px;
+    max-width: 100%;
```
Comment (lieut-data) [inline site/static/css/styles.css]: This makes the images readable on mobile.

File: site/content/blog/2019-06-04-percona-live-2019.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+* [Galera Cluster](http://galeracluster.com/products/): A sync, multi-master setup for MySQL and MariaDB (though increasingly focusing on the open-source MariaDB over MySQL).
+
+In addition to the stock options available, there is always the opportunity to tune the replication algorithm for a specific workload. [Anirban Rahut](https://www.linkedin.com/in/anirban-rahut-8973071/) from Facebook gave a session in which he spoke about their modifications to the (non-group) replication algorithm in MySQL. While the master copy can accepts writes in parallel, MySQL replicas currently apply all changes from the master copy serially. If the master copy receives a high volume of writes, the replication lag can grow uncontrollably, with the replicas unable to catch up. Anirban and his team modified the MySQL source to parse and interpret the replication stream, build a [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) modelling the write dependencies, and apply subsets of the binary logs in parallel, all while preserving commit order. For typical workloads, these replicas improved throughput by 4-5x!
```
Comment (amyblais) [inline site/content/blog/2019-06-04-percona-live-2019.md]:
```suggestion
In addition to the stock options available, there is always the opportunity to tune the replication algorithm for a specific workload. [Anirban Rahut](https://www.linkedin.com/in/anirban-rahut-8973071/) from Facebook gave a session in which he spoke about their modifications to the (non-group) replication algorithm in MySQL. While the master copy can accept writes in parallel, MySQL replicas currently apply all changes from the master copy serially. If the master copy receives a high volume of writes, the replication lag can grow uncontrollably, with the replicas unable to catch up. Anirban and his team modified the MySQL source to parse and interpret the replication stream, build a [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) modelling the write dependencies, and apply subsets of the binary logs in parallel, all while preserving commit order. For typical workloads, these replicas improved throughput by 4-5x!
```

Comment (amyblais) [reply site/content/blog/2019-06-04-percona-live-2019.md]: ("While the master copy can accepts..." - remove "s" in "accepts")

File: site/content/blog/2019-06-04-percona-live-2019.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+When a replica is configured, the server receiving the write is also trying to make a copy of the data. Asynchronous replication releases the client before the copy has been confirmed on the replica:
+
+![image](/blog/2019-06-04-percona-live-2019/transaction-2.svg)
+
+A power loss before the copy is made would result in the replica being out-of-date, but it can easily catch up from the master copy on restart. Even without a power loss, there is a period of time in which the replica will be out-of-date before it receives the copy. This period of time (sometimes measured as a number of bytes) is called the replication lag.
+
+When an application directly queries an asynchronous replica, it must be prepared to deal with some replication lag. For example, such an application should never `INSERT` a row on one server and then immediately query it from another: the data won't be there yet! A healthy database keeps replication lag to a minimum. If a replica fails to keep up, it might be necessary for the application to stop querying it to avoid unexpected results.
```
Comment (jwilander) [inline site/content/blog/2019-06-04-percona-live-2019.md]: What about adding a short tidbit about how we ran into this at Mattermost and solved it? I remember we had an issue with this on user and channel member creation before

Comment (lieut-data) [reply site/content/blog/2019-06-04-percona-live-2019.md]: Great idea! @jespino conveniently fixed just such an issue in v5.12.

**Pull request merged**